rpc/DatabaseRPC.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package grakn.client.rpc;
+
+import grakn.client.GraknClient;
+import grakn.protocol.DatabaseProto;
+import grakn.protocol.GraknGrpc;
+
+import static grakn.client.rpc.ClientRPC.rpcCall;
+import static grakn.client.rpc.DatabaseManagerRPC.nonNull;
+
+public class DatabaseRPC implements GraknClient.Database {
+
+    private final String name;
+    private final GraknGrpc.GraknBlockingStub blockingGrpcStub;
+
+    public DatabaseRPC(DatabaseManagerRPC databaseManager, String name) {
+        this.name = name;
+        this.blockingGrpcStub = databaseManager.blockingGrpcStub();
+    }
+
+    @Override
+    public String name() {
+        return name;
+    }
+
+    @Override
+    public void delete() {
+        rpcCall(() -> blockingGrpcStub.databaseDelete(DatabaseProto.Database.Delete.Req.newBuilder().setName(nonNull(name)).build()));
+    }
+
+    @Override
```

Comment:
Are we implementing `equals` and `hashCode` so that we can compare one RPC object to another?

What if we simply compare the name rather than the object itself? (eg., `rpc1.name().equals(rpc2.name())` instead of `rpc1.equals(rpc2)` 

When it comes to equality checks, I find it a bit too magical when implemented on complex objects such as RPC. 

Would an RPC object for the same database be considered the same object even if the underlying connection is different? Given that the object itself is called "RPC", it may be surprising that we actually do not consider the underlying RPC object for determining equality.

Reply:
Hmm. Yes I can see that that's weird. Let's not override equals here.

Reply:
Deleted `equals`, `hashCode`